### PR TITLE
Fix a few bugs in remove-compile-engine

### PR DESCRIPTION
**Pull request by csullivan**
1. Don't add relay main function to list of lowered TIR functions (codegen complains about only knowing how to handle a PrimFunc)

2. Don't skip visiting call to relay function in graph runtime codegen (otherwise some relay primitive function calls are not lowered to primfncalls and errors downstream in graph_runtime_codegen occur).

### CODE REVIEW

File: src/relay/backend/graph_runtime_codegen.cc [hunk 1]
```diff
@@ -254,8 +254,6 @@ class GraphRuntimeCodegen : public backend::MemoizedExprTranslator<std::vector<G
 
     ret.lowered_funcs = lowered_module.per_target_module;
     std::cout << "Modules: " << ret.lowered_funcs << std::endl;
```
Comment (zhiics) [inline src/relay/backend/graph_runtime_codegen.cc:256]: please remove this printing msg

Comment (zhiics) [inline src/relay/backend/graph_runtime_codegen.cc:256]: please removed this printing msg

Comment (csullivan) [reply src/relay/backend/graph_runtime_codegen.cc:256]: Hey @zhiics, thanks for checking this branch out! At the moment this branch is just a PoC @jroesch and I are working on. There's lots of cleanup we plan to batch together down the line a bit.

Comment (zhiics) [reply src/relay/backend/graph_runtime_codegen.cc:256]: oops.  I didn't notice the branch... Thanks for working on it

**Pull request merged**